### PR TITLE
BIN-65: Adopt database merge tool for migration

### PR DESCRIPTION
**Pull request by liodnik**

### CODE REVIEW

**Comment by liodnik** [comment]
@levon-vardevanyan-solarwinds , @ihar-khudzeyeu can you review this PR please?

File: mysql_merge/config.py.example [hunk 1]
```diff
@@ -90,3 +99,11 @@ preprocess_queries = []
...
 """
 default_mapping = { }
+
+"""
+  List of enum tables which should be equal in each db.
+  Ignored during merge.
+"""
+enum_tables = []
+
+ExecutionMode = ExecutionMode.DEFAULT
```
Comment (levon-vardevanyan-solarwinds) [inline mysql_merge/config.py.example]: not a big fan of using the same name for class and variable, why not `execution_mode`

File: mysql_merge/mysql_merger.py [hunk 2]
```diff
@@ -85,102 +81,103 @@ def merge(self):
 
         self._fk_checks(False)
 
-        self._logger.log(" -> 1/9 Executing preprocess_queries (specified in config)")
+        self._logger.log(" -> 1/11 Executing preprocess_queries (specified in config)")
```
Comment (levon-vardevanyan-solarwinds) [inline mysql_merge/mysql_merger.py:84]: what do you think about making this stuff variables and increment it after every stage?
not really important, but making one constant probably will save you eleven diff lines next time.

File: mysql_merge/mysql_merger.py [hunk 3]
```diff
@@ -189,10 +186,8 @@ def convert_mapped_fks_to_real_fks(self):
                     constraint_name, table_name, col_name), e, self._conn)
 
     def null_orphaned_fks(self):
-        cur = self._cursor
-
         # Null orphaned FKs
```
Comment (levon-vardevanyan-solarwinds) [inline mysql_merge/mysql_merger.py]: not sure how helpful this comment is

File: mysql_merge/mysql_merger.py [hunk 4]
```diff
@@ -329,35 +320,121 @@ def copy_data_to_target(self):
...
+
+        for table_name in self._config.enum_tables:
+            if not (table_name in self._db_map.keys() and table_name in self._destination_db_map.keys()):
+                self._logger.log(
+                    "----> '%(table)s' is missing on source or dest db" % {"table": table_name})
+                continue
+
+            columns = self._source_mapper.get_overlapping_columns(
+                self._destination_db_map, table_name)
+            if not len(columns):
```
Comment (levon-vardevanyan-solarwinds) [inline mysql_merge/mysql_merger.py:380]: what is the type of `columns`? thought it is either `if len(columns) != 0:` or just `if columns:`

also matter of taste I guess:
`on += "" if len(on) == 0 else " AND"` <-> `on += "" if not on else " AND"`
`where += "" if len(where) == 0 else " OR"` <-> `where += "" if not where else " OR"`

File: mysql_merge/mysql_merger.py [hunk 5]
```diff
@@ -329,35 +320,121 @@ def copy_data_to_target(self):
...
+                handle_exception("Exception occured while comparing '%(t)s' enum table" % {
+                                 "t": table_name}, e, self._conn)
+
+            self._logger.log(
+                "----> Enum Table %s is compatible with destination database and will be ignored during merge" % table_name)
+            self._db_map.pop(table_name)
+            self._destination_db_map.pop(table_name)
+
+    def get_increment_value(self, table_name):
+        if table_name in self._config.increment_step:
```
Comment (levon-vardevanyan-solarwinds) [inline mysql_merge/mysql_merger.py:430]: :D (ignore me)

`increment_step = self._config.increment_step[table_name if table_name in self._config.increment_step else 'default']`

**Review by levon-vardevanyan-solarwinds — APPROVED** [review]
lgtm, probably mixing together two styles of string formatting (`%` vs `.format`) isn't the most beautiful thing in the world and to me function `handle_enum_tables` is a little messy and hard to read, but overall looks fine.

**Pull request merged**